x/auth/vesting/types/vesting_account.go

```diff
@@ -566,3 +566,69 @@ func (dva DelayedVestingAccount) String() string {
 	out, _ := dva.MarshalYAML()
 	return out.(string)
 }
+
+//-----------------------------------------------------------------------------
+// Permanent Locked Vesting Account
+
+var _ vestexported.VestingAccount = (*PermanentLockedVestingAccount)(nil)
+var _ authtypes.GenesisAccount = (*PermanentLockedVestingAccount)(nil)
+
+// NewPermanentLockedVestingAccountRaw creates a new PermanentLockedVestingAccount object from BaseVestingAccount
+func NewPermanentLockedVestingAccountRaw(bva *BaseVestingAccount) *PermanentLockedVestingAccount {
+	// ensure EndTime is set to -1, as PermanentLockedVestingAccount's do not have an EndTime
+	bva.EndTime = -1
+
+	return &PermanentLockedVestingAccount{
+		BaseVestingAccount: bva,
+	}
+}
+
+// NewPermanentLockedVestingAccount returns a PermanentLockedVestingAccount
+func NewPermanentLockedVestingAccount(baseAcc *authtypes.BaseAccount, originalVesting sdk.Coins) *PermanentLockedVestingAccount {
+	baseVestingAcc := &BaseVestingAccount{
+		BaseAccount:     baseAcc,
+		OriginalVesting: originalVesting,
+		EndTime:         -1,
+	}
+
+	return &PermanentLockedVestingAccount{baseVestingAcc}
+}
+
+// GetVestedCoins returns the total amount of vested coins for a permanent locked vesting
+// account. All coins are only vested once the schedule has elapsed.
+func (plva PermanentLockedVestingAccount) GetVestedCoins(blockTime time.Time) sdk.Coins {
+	return nil
+}
+
+// GetVestingCoins returns the total number of vesting coins for a permanent locked
+// vesting account.
+func (plva PermanentLockedVestingAccount) GetVestingCoins(blockTime time.Time) sdk.Coins {
+	return plva.OriginalVesting
+}
+
+// LockedCoins returns the set of coins that are not spendable (i.e. locked).
+func (plva PermanentLockedVestingAccount) LockedCoins(blockTime time.Time) sdk.Coins {
+	return plva.BaseVestingAccount.LockedCoinsFromVesting(plva.GetVestingCoins(blockTime))
```

Comment:
```suggestion
	return nil
```

Reply:
Your suggestion is incorrect imo, `LockedCoins` means vesting coins that are not delegated.